standalone-metastore/metastore-catalog/src/main/java/org/apache/iceberg/rest/HMSCatalogAdapter.java

```diff
@@ -0,0 +1,778 @@
+import org.apache.iceberg.rest.requests.UpdateNamespacePropertiesRequest;
+import org.apache.iceberg.rest.requests.UpdateTableRequest;
+import org.apache.iceberg.rest.responses.ConfigResponse;
+import org.apache.iceberg.rest.responses.CreateNamespaceResponse;
+import org.apache.iceberg.rest.responses.ErrorResponse;
+import org.apache.iceberg.rest.responses.GetNamespaceResponse;
+import org.apache.iceberg.rest.responses.ListNamespacesResponse;
+import org.apache.iceberg.rest.responses.ListTablesResponse;
+import org.apache.iceberg.rest.responses.LoadTableResponse;
+import org.apache.iceberg.rest.responses.LoadViewResponse;
+import org.apache.iceberg.rest.responses.OAuthTokenResponse;
+import org.apache.iceberg.rest.responses.UpdateNamespacePropertiesResponse;
+import org.apache.iceberg.util.Pair;
+import org.apache.iceberg.util.PropertyUtil;
+
+/**
+ * Original @ https://github.com/apache/iceberg/blob/main/core/src/test/java/org/apache/iceberg/rest/RESTCatalogAdapter.java
+ * Adaptor class to translate REST requests into {@link Catalog} API calls.
+ */
+public class HMSCatalogAdapter implements RESTClient {
+  /**  The metric names prefix. */
+  static final String HMS_METRIC_PREFIX = "hmscatalog.";
+  private static final Splitter SLASH = Splitter.on('/');
+
+  private static final Map<Class<? extends Exception>, Integer> EXCEPTION_ERROR_CODES =
+      ImmutableMap.<Class<? extends Exception>, Integer>builder()
+          .put(NamespaceNotSupported.class, 400)
+          .put(IllegalArgumentException.class, 400)
+          .put(ValidationException.class, 400)
+          .put(NamespaceNotEmptyException.class, 400) // TODO: should this be more specific?
+          .put(NotAuthorizedException.class, 401)
+          .put(ForbiddenException.class, 403)
+          .put(NoSuchNamespaceException.class, 404)
+          .put(NoSuchTableException.class, 404)
+          .put(NoSuchIcebergTableException.class, 404)
+          .put(UnsupportedOperationException.class, 406)
+          .put(AlreadyExistsException.class, 409)
+          .put(CommitFailedException.class, 409)
+          .put(UnprocessableEntityException.class, 422)
+          .put(CommitStateUnknownException.class, 500)
+          .buildOrThrow();
+
+  private final Catalog catalog;
+  private final SupportsNamespaces asNamespaceCatalog;
+  private final ViewCatalog asViewCatalog;
+
+
+  public HMSCatalogAdapter(Catalog catalog) {
+    this.catalog = catalog;
+    this.asNamespaceCatalog =
+        catalog instanceof SupportsNamespaces ? (SupportsNamespaces) catalog : null;
+    this.asViewCatalog = catalog instanceof ViewCatalog ? (ViewCatalog) catalog : null;
+  }
+
+  enum HTTPMethod {
+    GET,
+    HEAD,
+    POST,
+    DELETE
+  }
+  
+  @FunctionalInterface
+  interface Operation<T extends RESTResponse>  {
+      T exec(Map<String, String> vars, Object body);
+  }
+
+  enum Route {
+    TOKENS(HTTPMethod.POST, "v1/oauth/tokens", null, OAuthTokenResponse.class),
```

Comment:
This routes are defined in Iceberg already: 
https://github.com/apache/iceberg/blob/31e9dc768cef47df5eb007577f25ef3bbff86f61/core/src/main/java/org/apache/iceberg/rest/Endpoint.java#L36

E.g: Polaris (another Iceberg REST Server app) using it. Please use this approach:
https://github.com/apache/polaris/blob/5244c4c0bbbe89e89d1b98dda65e2e89eb970732/service/common/src/main/java/org/apache/polaris/service/catalog/IcebergCatalogAdapter.java#L84

Reply:
The Endpoint(s) are not defining how to extract the variables from the url which is the point of these Route(s).